Lib/_pyrepl/readline.py

```diff
@@ -457,9 +467,12 @@ def read_history_file(self, filename: str = gethistoryfile()) -> None:
     def write_history_file(self, filename: str = gethistoryfile()) -> None:
         maxlength = self.saved_history_length
         history = self.get_reader().get_trimmed_history(maxlength)
-        f = open(os.path.expanduser(filename), "w",
-                 encoding="utf-8", newline="\n")
-        with f:
+
+        filename = os.path.expanduser(filename)
+        is_editline = self._is_editline_history(filename)
+        with open(filename, "w", encoding="utf-8", newline="\n") as f:
+            if is_editline:
+                f.write(f"{_EDITLINE_MARKER}\n")
```

Comment:
This is incorrect and this change is irrelevant to the bug you're fixing. The editline format is somewhat different, a file written by us will not be readable by editline anyway, and now you'll make pyrepl mangle unicode in its own history until you delete the history file.